### PR TITLE
Docs: update release process.

### PR DESCRIPTION
**Pull request by godai78**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Docs: update release process. Closes https://github.com/cksource/ckeditor5-internal/issues/3417

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

**Review by pomek — APPROVED** [review]
LGTM.

File: docs/updating/release-process.md [hunk 1]
```diff
@@ -7,22 +7,34 @@ meta-description: Learn how the CKEditor 5 versions are published.
...
 
 It is good to follow npm release messages about new packages being published as well as periodically check the changelog.
 
-## Public release
+### Nightly releases
 
-Soon after the code release, a more user-oriented information is published. In a [release blog post](https://ckeditor.com/blog/?category=releases&tags=CKEditor-5) the latest release version is presented in more details, with examples, screencasts, screenshots and links to the new or updated documentation if it brings important information about the release. The release blog post would also provide additional information and context for the changes if needed and direct the user toward migration guides if these were created for the release.
+The `nightly` releases are published daily and are built upon the current codebase. It means they may contain new features and bugfixes, but at the same time &ndash; mistakes and errors. Because of that, `nightly` versions are not fit for production environments. They should be seen more as an experimental approach or testing grounds.
```
Comment (Witoso) [inline docs/updating/release-process.md:24]: > the current codebase's _master branch_.

we can be more explicit here.

File: docs/updating/release-process.md [hunk 2]
```diff
@@ -7,22 +7,34 @@ meta-description: Learn how the CKEditor 5 versions are published.
...
 
-It is good to follow [CKEditor Ecosystem Blog](https://ckeditor.com/blog/) as it also brings other important articles and news about features, changes and planned development. You can also [sign up to the monthly newsletter](https://ckeditor.com/newsletter/) in order to be notified about the latest releases.
+### Alpha and RC releases
+
+Sometimes, to test specific new features or important in the upcoming stable release, `alpha` versions are released, too. While they would typically be based on a tested code, they are still not `stable` and hence the same caution is advised when trying these out.
+
+The release candidate (`rc`) versions are often the ones that will become `stable` during the upcoming release.
+
+The `rc` and `alpha` versions are named by appending a numbered type denominator to the previous stable version. The `nightly` ones are marked with a daily date.
```
Comment (Witoso) [inline docs/updating/release-process.md:32]: > Releases will be created daily, so integration tests using nightly packages might fail. **Please create an issue in the CKEditor 5 issue tracker in such a case. We may have merged a regression, and our tests did not catch it.**

I like :point\_up:  @pomek wrote in the issue. We could try to incorporate it here.

File: docs/updating/release-process.md (hunk 1 above)
Comment (godai78) [reply docs/updating/release-process.md:24]: In commit.

File: docs/updating/release-process.md (hunk 2 above)
Comment (godai78) [reply docs/updating/release-process.md:32]: Added to the  Nightly section in commit.

**Pull request merged**